File: 162. Find Peak Element/findPeakElement.py
from typing import List

class Solution:
  def findPeakElement(self, nums: List[int]) -> int:
    left = 0
    right = len(nums) - 1
    
    while left < right:
      mid = (left + right) // 2
      if nums[mid + 1] >= nums[mid]:
        left = mid + 1
      else:
        right = mid
        
    return left
    
    
nums = [1, 2, 3, 1]
solution = Solution()
print('findPeakElement: ', solution.findPeakElement(nums))

File: 162. Find Peak Element/findPeakElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const findPeakElement = (nums) => {
  let left = 0;
  let right = nums.length - 1;

  while (left < right) {
    const mid = Math.floor((left + right) / 2);
    if (nums[mid + 1] >= nums[mid]) {
      left = mid + 1;
    } else {
      right = mid;
    }
  }

  return left;
};


const nums = [1, 2, 3, 1];
console.log('findPeakElement: ', findPeakElement(nums));
